File: src/main/java/com/animalfarm/admin/service/TradeHistoryService.java
package com.animalfarm.admin.service;

import com.animalfarm.admin.collection.TradeHistory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface TradeHistoryService {
    Page<TradeHistory> getPage(Pageable pageable, String keyword);
}

File: src/main/kotlin/com/animalfarm/admin/service/TradeHistoryService.kt
package com.animalfarm.admin.service

import com.animalfarm.admin.domain.TradeHistoryDTO
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface TradeHistoryService {
    fun getPage(pageable: Pageable, keyword: String): Page<TradeHistoryDTO>
}